src/main/java/com/epam/training/designpatterns/fastfoodrestaurant/Chef.java

```diff
@@ -0,0 +1,58 @@
+package com.epam.training.designpatterns.fastfoodrestaurant;
+
+import java.util.Queue;
+
+import com.epam.training.designpatterns.fastfoodrestaurant.Menu.ExtraType;
+import com.epam.training.designpatterns.fastfoodrestaurant.Menu.FoodType;
+
+public class Chef {
+    private Queue<Order> orders;
+    private Food food;
+
+
+    public Chef(Queue<Order> orders) {
+        this.orders = orders;
+    }
+
+    public void makeFood() {
+        Order order = orders.remove();
+        System.out.println("Chef: Getting order from queue.");
+        
+        FoodBuilder foodBuilder = new FoodBuilder(order.getFood());
+        foodBuilder.addExtra(order.getExtra());
+        food = foodBuilder.makeFood();
+        System.out.println("Chef: I made the food on my secret food making machine.");
+
+        System.out.println("Chef: The order is done.");
+        order.setDone();        
```

Comment:
Működik, de valahogy jobb lenne, ha a szakács nem tudna ilyet csinálni, ő csak a pultosnak tudna szólni és a pultos tudná elérni a vendéget.

Mintha sérülne a Law of Demeter elv, miszerint nem beszélünk idegenekkel, illetve a szépen megvalósított mediator patternt is megkerüli ha belegondolunk, mert a szakács mégis csak tud közvetlenül szólni a vendéghez.

https://en.wikipedia.org/wiki/Law_of_Demeter

Ez kicsit szürke zóna, mert alkalmazástól is függ, hogy hány rétegre van szükség, lehet, hogy egy ilyen lejuttatott "token" segítségével, ami itt az `order` példány, akarunk valamilyen eseményt kiváltani. Szóval ez jó is lehet így, inkább csak gondolatébresztőnek szántam ezt a pár sort.